tip-distributor/src/lib.rs

```diff
@@ -460,104 +470,212 @@ pub fn derive_tip_distribution_account_address(
     )
 }
 
+pub const MAX_RETRIES: usize = 5;
+pub const FAIL_DELAY: Duration = Duration::from_millis(100);
+
+/// Returns unprocessed transactions, along with fail count
+pub async fn sign_and_send_transactions_with_retries_multi_rpc(
+    signer: &Arc<Keypair>,
+    blockhash_rpc_client: &Arc<RpcClient>,
+    rpc_clients: &Arc<Vec<Arc<RpcClient>>>,
+    mut transactions: Vec<Transaction>,
+    max_loop_duration: Duration,
+) -> (Vec<Transaction>, u64) {
+    let error_count = Arc::new(AtomicU64::default());
+    let blockhash = Arc::new(RwLock::new(
+        blockhash_rpc_client
+            .get_latest_blockhash()
+            .await
+            .expect("fetch latest blockhash"),
+    ));
+    let mut rng = rand::thread_rng();
+    transactions.shuffle(&mut rng); // shuffle to avoid sending same txns as other claim-tip processes
```

Comment:
comment somewhat misleading, will still send the same? and have potential for failed txs?

Reply:
shuffle is mainly if we run both warehouse nodes, would make them less likely to race for the same txns.

Reply:
Still has potential for failed txns, but we refetch all accounts to see if they're claimed every couple mins to avoid duplicate work.